packages/driver/src/cypress/log.js

```diff
@@ -155,9 +156,21 @@ const defaults = function (state, config, obj) {
       },
     })
 
-    // if obj.isCurrent
-    // stringify the obj.message (if it exists) or current.get("args")
-    obj.message = $utils.stringify(obj.message != null ? obj.message : (current != null ? current.get('args') : undefined))
+    if (obj.message || obj.message === '') {
+      obj.message = $utils.stringify(obj.message)
+    } else {
+      let args = current != null ? current.get('args') : undefined
+
+      if (args) {
+        const optionsIndex = getIndexOfUserOptions(current.get('name'), args)
+
+        if (optionsIndex !== -1 && optionsIndex < args.length) {
+          args = args.filter((v, i) => i !== optionsIndex)
+        }
+      }
```

Comment:
I think this could have clearer comments / variable names explaining what we're doing here.

Reply:
Added comment below:

```
// Sometimes, option argument isn't at the end of the list.
// The code below removes that option argument from the list and
// generates message.
```